rali/rali_pybind/example/README.md

```diff
@@ -0,0 +1,110 @@
+```
+
+### RALI-TensorFlow training on pets dataset
+This example uses the RALI TFRecordReader to read images from the disk, perform the necessary augmentations and run a classification training on the pets dataset.
+```
+cd /root/tf_petsTrainingExample
+```
+
+This example can also be obtained by cloning this repository inside the docker container. This "tf_petsTrainingExample folder is present in the repository under:"
+```
+cd rali/rali_pybind/example/tf_petsTrainingExample
+```
+
+For first run, to setup the dataset, edit "train_withRALI_withTFRecordReader.py" and set "DATASET_DOWNLOAD_AND_PREPROCESS = True"
+For subsequent runs, after the dataset has already been downloaded and preprocessed, set "DATASET_DOWNLOAD_AND_PREPROCESS = False"
+
+```
+python3 train_withRALI_withTFRecordReader.py
+```
+
+## Using RALI TFRecordReader
+For using RALI TFRecordReader for image classification or detection purposes, please follow the standard below.
+
+For image classification, RALI needs the following 3 features in the user's TFRecord to be read, as seen in rali/pybind_python/example/tf_classification.py
+```
+features = 
+{
+    'image/encoded':tf.FixedLenFeature((), tf.string, ""),
+    'image/class/label':tf.FixedLenFeature([1], tf.int64,  -1),
+    'image/filename':tf.FixedLenFeature((), tf.string, "")
+}
+```
+
+For object detection, RALI needs the following 8 features in the user's TFRecord to be read, as seen in rali/pybind_python/example/tf_detection.py
+```
+features = 
+{
+    'image/encoded':tf.FixedLenFeature((), tf.string, ""),
+    'image/class/label':tf.FixedLenFeature([1], tf.int64,  -1),
+    'image/class/text':tf.FixedLenFeature([ ], tf.string, ""),
+    'image/object/bbox/xmin':tf.VarLenFeature(dtype=tf.float32),
+    'image/object/bbox/ymin':tf.VarLenFeature(dtype=tf.float32),
+    'image/object/bbox/xmax':tf.VarLenFeature(dtype=tf.float32),
+    'image/object/bbox/ymax':tf.VarLenFeature(dtype=tf.float32),
+    'image/filename':tf.FixedLenFeature((), tf.string, "")
+}
+```
+
+Since the key names in your dataset's TFRecords might vary from those expected by RALI's TFRecordReader, please provide the following 'featureKeyMap' that maps your dataset's TFRecord key names to those expected by the TFRecordReader:
+
+### Format:
+```
+featureKeyMap = 
+{
+    <Key name in RALI TFRecordReader (the 3/8 keys above for classification/detection)> : <Corresponding key name in your dataset's TFRecords>,
+    <Key name in RALI TFRecordReader (the 3/8 keys above for classification/detection)> : <Corresponding key name in your dataset's TFRecords>,
+}
+```
+
+### Possible example for Image Classification:
+```
+featureKeyMap = 
+{
+    'image/encoded':'encoded',
+    'image/class/label':'IDs',
+    'image/filename':'filename'
+}
+```
+or
+```
+featureKeyMap = 
+{
+    'image/encoded':'image/encoded',
+    'image/class/label':'image/class/IDs',
+    'image/filename':'image/filename'
+}
+```
+
+### Possible example for Object Detection:
```

Comment:
Codacy found an issue: [[no-heading-punctuation] Don’t add a trailing `:` to headings](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5976200)